src/lib/esnext.array.d.ts

```diff
@@ -0,0 +1,104 @@
+interface Array<T> {

```

Comment:
Just in case there are any existing `--lib esnext.array` options in the wild, I would recommend `esnext.array` reference `es2019.array`.

```suggestion
/// <reference lib="es2019.array" />
interface Array<T> {
```